tests/van/test_modeling_van.py

```diff
@@ -0,0 +1,280 @@
+        self.config_tester.create_and_test_config_with_num_labels()
+        self.config_tester.check_config_can_be_init_without_params()
+        self.config_tester.check_config_arguments_init()
+
+    def create_and_test_config_common_properties(self):
+        return
+
+    @unittest.skip(reason="Van does not use inputs_embeds")
+    def test_inputs_embeds(self):
+        pass
+
+    @unittest.skip(reason="Van does not support input and output embeddings")
+    def test_model_common_attributes(self):
+        pass
+
+    def test_forward_signature(self):
+        config, _ = self.model_tester.prepare_config_and_inputs_for_common()
+
+        for model_class in self.all_model_classes:
+            model = model_class(config)
+            signature = inspect.signature(model.forward)
+            # signature.parameters is an OrderedDict => so arg_names order is deterministic
+            arg_names = [*signature.parameters.keys()]
+
+            expected_arg_names = ["pixel_values"]
+            self.assertListEqual(arg_names[:1], expected_arg_names)
+
+    def test_model(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_model(*config_and_inputs)
+
+    def test_initialization(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+        configs_no_init = _config_zero_init(config)
+
+        for model_class in self.all_model_classes:
+            model = model_class(config=configs_no_init)
+            for name, module in model.named_modules():
+                if isinstance(module, (nn.BatchNorm2d, nn.GroupNorm, nn.LayerNorm)):
+                    self.assertTrue(
+                        torch.all(module.weight == 1),
+                        msg=f"Parameter {name} of model {model_class} seems not properly initialized",
+                    )
+                    self.assertTrue(
+                        torch.all(module.bias == 0),
+                        msg=f"Parameter {name} of model {model_class} seems not properly initialized",
+                    )
+                elif isinstance(module, nn.Conv2d):
+                    # fan_out = module.kernel_size[0] * module.kernel_size[1] * module.out_channels
+                    # fan_out //= module.groups
+                    # std = math.sqrt(2.0 / fan_out)
+                    # we should use shapiro test https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.shapiro.html
+                    # but I can't import scipy
+                    # # impossibile to have numbers outside 6 x std
+                    # normal_range = [-6 * std, 6 * std]
+                    # self.assertTrue(torch.all(module.weight.data >= normal_range[0]))
+                    # self.assertTrue(torch.all(module.weight.data <= normal_range[1]))
+                    # # check bias
+                    # self.assertTrue(torch.all(module.bias == 0))
```

Comment:
To finish?

Reply:
To discuss, in order to properly test that the weights of `nn.Conv2d`s were initialized by `torch.nn.init.normal_` I need to check the weights were sampled by a normal distribution with that mean and that std. To do so I need to perform what is called a shapiro test that is implemented in scipy but I don't think I can import.

If that is okay I can just skip the test for `nn.Conv2d`

Reply:
I think we can add the scipy dep for testing if it's useful here.

Reply:
What is the correct one to use `scipy` in a test file? Can I just import it?

Reply:
Resolved!